File: app/backend (1)/backend/src/main/java/com/naariraksha/backend/service/UserService.java
package com.naariraksha.backend.service;

import com.naariraksha.backend.dto.LoginRequest;
import com.naariraksha.backend.dto.UserRegisterRequest;
import com.naariraksha.backend.entity.User;
import com.naariraksha.backend.repository.UserRepository;
import org.springframework.stereotype.Service;
import java.util.Map;
import java.util.Optional;

@Service
public class UserService {

    private final UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public String registerUser(UserRegisterRequest request) {
        if (userRepository.findByEmail(request.getEmail()).isPresent()) {
            throw new RuntimeException("Email already registered!");
        }

        User user = User.builder()
                .name(request.getName())
                .email(request.getEmail())
                .password(request.getPassword()) // In production, use BCrypt!
                .build();

        userRepository.save(user);
        return "User " + request.getName() + " registered successfully";
    }

    public Map<String, Object> loginUser(LoginRequest request) {
        Optional<User> userOpt = userRepository.findByEmail(request.getEmail());
        
        if (userOpt.isPresent() && userOpt.get().getPassword().equals(request.getPassword())) {
            User user = userOpt.get();
            return Map.of(
                "status", "success",
                "message", "Login successful",
                "userId", user.getId(),
                "name", user.getName()
            );
        }
        
        return Map.of("status", "error", "message", "Invalid email or password");
    }
}

File: backend/backend/src/main/kotlin/com/naarisafe/backend/service/UserService.kt
package com.naarisafe.backend.service

import com.naarisafe.backend.entity.User
import com.naarisafe.backend.repository.UserRepository
import org.springframework.stereotype.Service

@Service
class UserService(
    private val userRepository: UserRepository
) {

    fun register(name: String, email: String, password: String) {
        if (userRepository.findByEmail(email) != null) {
            throw IllegalArgumentException("Email already registered")
        }

        val user = User(
            name = name,
            email = email,
            password = password   // hashing later
        )

        userRepository.save(user)
    }

    fun login(email: String, password: String): User {
        val user = userRepository.findByEmail(email)
            ?: throw IllegalArgumentException("Invalid email or password")

        if (user.password != password) {
            throw IllegalArgumentException("Invalid email or password")
        }

        return user
    }
}
